### PR TITLE
Enable updates for ScalableTarget and ScalingPolicy [on branch update-runtime-0.12]

### PR DESCRIPTION
**Pull request by mbaijal**
Issue #, if available:
https://github.com/aws-controllers-k8s/community/issues/867

Description of changes:
1. Enable updates based on changes to code generator made in this PR - https://github.com/aws-controllers-k8s/code-generator/pull/166
2. Add an integration test to check for updates

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by surajkota — COMMENTED** [review]
little hard to review with all the runtime and codegen changes, but on a high level looks good

need to replace sleep with a more reliable way to wait for the update, -> which is already planned

will review again once #28  is merged

**Comment by ryansteakley** [comment]
> little hard to review with all the runtime and codegen changes, but on a high level looks good
> 
> need to replace sleep with a more reliable way to wait for the update, -> which is already planned
> 
> will review again once #28 is merged

Is the more reliable way to wait for update planned for this pr? @mbaijal  Otherwise lgtm

File: pkg/resource/scaling_policy/manager_test_suite_test.go [hunk 1]
```diff
@@ -17,38 +17,38 @@ import (
...
-     return &resourceManager{
-     	    rr:           nil,
-	    awsAccountID: "",
-	    awsRegion:    "",
-	    sess:         nil,
-	    sdkapi:       mockApplicationAutoScalingAPI,
-	    log:          fakeLogger,
-	    metrics:      ackmetrics.NewMetrics("applicationautoscaling"),
-     }
+	zapOptions := ctrlrtzap.Options{
```
Comment (akartsky) [inline pkg/resource/scaling_policy/manager_test_suite_test.go:38]: nit: can we replace all of these tabs with spaces instead in all of the test_suit related files

**Review by akartsky — APPROVED** [review]

File: pkg/resource/scaling_policy/manager_test_suite_test.go (hunk 1 above)
Comment (mbaijal) [reply pkg/resource/scaling_policy/manager_test_suite_test.go:38]: both testfiles and testdata have tabs at the moment, why do we need to change to space ?

**Comment by mbaijal** [comment]
Created a new PR #32

**Pull request closed without merging**